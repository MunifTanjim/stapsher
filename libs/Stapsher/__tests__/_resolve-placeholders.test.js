const helpers = require('../../../__tests__/helpers')

helpers.addSnapshotSerializers()

helpers.mockDate()
helpers.mockUUIDv1()

const utils = require('../utils')
utils.resolvePlaceholders = jest.fn(() => 'resolved_string')

const Stapsher = require('../../Stapsher')

const string = 'string_with_{placeholders}'

describe('libs/Stapsher:__resolvePlaceholders', () => {
  const parameters = helpers.getParameters()

  let stapsher
  beforeEach(() => {
    utils.resolvePlaceholders.mockClear()

    stapsher = new Stapsher(parameters)

    stapsher.fields = { ...helpers.getFields() }
    stapsher.options = { ...helpers.getOptions() }
  })

  it('sets the dictionary if it does not exist', () => {
    expect(stapsher.dictionary).toBeNull()

    stapsher.__resolvePlaceholders(string)

    expect(stapsher.dictionary).not.toBeNull()
    expect(stapsher.dictionary).toMatchSnapshot()
  })

  it('does not set the dictionary if it exists', () => {
    let dictionary = { exists: true }
    stapsher.dictionary = { ...dictionary }

    expect(stapsher.dictionary).not.toBeNull()

    stapsher.__resolvePlaceholders(string)

    expect(stapsher.dictionary).toEqual(dictionary)
  })

  it('calls necessary functions', () => {
    stapsher.__resolvePlaceholders(string)

    expect(utils.resolvePlaceholders).toHaveBeenCalledTimes(1)
    expect(utils.resolvePlaceholders).toHaveBeenCalledWith(
      string,
      stapsher.dictionary
    )
  })

  it('returns resolved string', () => {
    expect(stapsher.__resolvePlaceholders(string)).toBe(
      utils.resolvePlaceholders()
    )
  })
})
