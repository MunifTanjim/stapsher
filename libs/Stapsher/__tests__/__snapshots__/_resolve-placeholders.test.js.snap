// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`libs/Stapsher:__resolvePlaceholders sets the dictionary if it does not exist 1`] = `
{
  "_id": "7w0.z3r0.7w0.53v3n",
  "_date": "1970-01-01T00:00:00.000Z",
  "fields": {
    "author": "Samantha Groves",
    "email": "samantha@example.com",
    "url": "https://samantha.example.com",
    "content": "We might as well be a symphony"
  },
  "options": {
    "alias": "Root"
  }
}
`;
